Add beeswarm plot to SHAP-IQ

annprzy opened the pull request:
This PR addresses #399, adapting SHAP beeswarm plot to handle interactions and adding it to the `plot` module.

New features:
- `shapiq/plot/beeswarm.py`: adapts SHAP beeswarm plot to handle interactions
- `tests/tests_plots/test_beeswarm.py`: adds test for beeswarm plot

Updated:
- `shapiq/__init__.py`: adds beeswarm plot to the list of imports and `__all__` attribute
- `shapiq/plot/__init__.py`: adds beeswarm plot to the list of imports and `__all__` attribute

mmschlk commented:
@annprzy, would it technically be possible to work without the skimage import? I see that you only require the conversion of color spaces there. I would love to not blow up the dependencies of shapiq too much and we already use [`colour`](https://pypi.org/project/colour/) or `matplotlib` which probably can also do what you need to do with skimage. Maybe the code would also work without the conversions even?

Second, I know you asked this in the issue, but if you do have some additional time for this, would you be able to add a quick tutorial for the plot under `api_examples` like the [upset plot](https://github.com/mmschlk/shapiq/blob/main/examples/api_examples/visualization_upset_plot.ipynb)? This really does not need to be too elaborate but some quick showcase of the plot? 

Thank you for this PR! :)

annprzy commented:
@mmschlk I simplified the colormap generation (skimage is not longer needed) and added the tutorial. Let me know your thoughts on this and whether any changes are needed.
